lib/Runtime/Types/DynamicObjectPropertyEnumerator.cpp

```diff
@@ -133,6 +140,13 @@ namespace Js
 
     bool DynamicObjectPropertyEnumerator::CanUseJITFastPath() const
     {
+#if ENABLE_TTD
+        if(this->scriptContext->ShouldPerformRecordOrReplayAction())
+        {
+            return false;
+        }
+#endif
```

Comment:
Is JIT enabled for TTD?  If so, you will have to look at Lowerer::GenerateInitForInEnumeratorFastPath